bites_test.py

```diff
@@ -1253,43 +1294,268 @@ def test_upon_initialisation_anthill_has_one_of_each_type_of_food(self):
+    anthill_rule = ""
+    wine_rule = ""
+    bites_game = Bites(ants, standard_tokens_for_trail, wine_tokens_for_trail, chocolate_tokens_for_trail, players, anthill_rule, wine_rule)
+    trail = ["cheese", "cheese", "bread", "cheese"]
+    expected_new_trail = ["cheese", "cheese", "bread", "cheese", "chocolate"]
+
+    with patch('bites.random.shuffle') as mock_random_shuffle:
+      actual_new_trail = bites_game.add_chocolate_into_trail(trail, chocolate_tokens_for_trail)
+      mock_random_shuffle.assert_called_once_with(["chocolate"])
+
+    self.assertEqual(actual_new_trail[0], "cheese")
+    self.assertEqual(actual_new_trail[1], "cheese")
+    self.assertEqual(actual_new_trail[2], "bread")
+    self.assertEqual(actual_new_trail[3], "cheese")
+    self.assertEqual(actual_new_trail, expected_new_trail)
+
+  @patch('bites.Bites.identify_chocolate_limit', return_value = 9)
+  def test_a_typical_example_of_a_game_trail_with_3_of_each_standard_and_2_choc_and_choc_lim_is_8(
+    self,
+    mock_choc_lim,
+    ):
+    # test 191
+    ants = []
+    standard_tokens_for_trail = {
+      "cheese": 3, 
+      "bread": 3,
+      "grapes": 3,
+      "apple": 3,
+      "pepper": 3,
+      }
+    wine_tokens_for_trail = {"wine": 2}
+    chocolate_tokens_for_trail = {"chocolate": 2}
+    players = []
+    anthill_rule = ""
+    wine_rule = ""
+    bites_game = Bites(ants, standard_tokens_for_trail, wine_tokens_for_trail, chocolate_tokens_for_trail, players, anthill_rule, wine_rule)
+    trail = [
+      "cheese",
+      "grapes",
+      "cheese",
+      "wine",
+      "pepper",
+      "cheese",
+      "bread",
+      "apple",
+      "pepper",
+      "bread",
+      "pepper",
+      "bread",
+      "grapes",
+      "apple",
+      "grapes",
+      "wine",
+      "apple",
+      ]
+    expected_new_trail_0_to_9 = [
+      "cheese",
+      "grapes",
+      "cheese",
+      "wine",
+      "pepper",
+      "cheese",
+      "bread",
+      "apple",
+      "pepper",
+    ]
+
+    with patch('bites.random.shuffle') as mock_random_shuffle:
+      actual_new_trail = bites_game.add_chocolate_into_trail(trail, chocolate_tokens_for_trail)
+      mock_random_shuffle.assert_called_once_with([
+        "bread",
+        "pepper",
+        "bread",
+        "grapes",
+        "apple",
+        "grapes",
+        "wine",
+        "apple",
+        "chocolate",
+        "chocolate",
+        ])
+    
+    self.assertEqual(actual_new_trail[0:9], expected_new_trail_0_to_9)
+    self.assertEqual(len(actual_new_trail), 19)
+    self.assertEqual(actual_new_trail.count("apple"), 3)
+    self.assertEqual(actual_new_trail.count("grapes"), 3)
+    self.assertEqual(actual_new_trail.count("cheese"), 3)
+    self.assertEqual(actual_new_trail.count("pepper"), 3)
+    self.assertEqual(actual_new_trail.count("bread"), 3)
+    self.assertEqual(actual_new_trail.count("chocolate"), 2)
+    self.assertEqual(actual_new_trail.count("wine"), 2)
+
+class InitialiseTrailTest(unittest.TestCase):
```

Comment:
The mocking in the tests in this class is a bit chaotic. I suggest we review all of these tests together in person.